crates/rpc/rpc/src/eth/api/transactions.rs

```diff
@@ -20,6 +21,77 @@ where
         unimplemented!()
     }
 
+    /// Finds a given trasaction by its hash.
+    pub(crate) async fn transaction_by_hash(
+        &self,
+        hash: H256,
+    ) -> EthResult<Option<reth_rpc_types::Transaction>> {
+        let tx_by_hash = self.client().transaction_by_hash(hash)
+            .map_err(|_| EthApiError::EmptyRawTransactionData)?
+            .ok_or(EthApiError::EmptyRawTransactionData)?
+            .into_ecrecovered()
+            .ok_or(EthApiError::InvalidTransactionSignature)?;
+
+        Ok(Some(Transaction::from_recovered_with_block_context(
+            tx_by_hash,
+            // this is just stubbed out for now still need to fully implement
+            H256::default(),
+            u64::default(),
+            U256::from(usize::from(Index::default())),
+        )))
+    }
+
+    /// Find a transaction at a specific index in a block, after finding the block by its hash.
+    pub(crate) async fn transaction_by_block_hash_and_index(
+        &self,
+        hash: H256,
+        index: Index,
+    ) -> EthResult<Option<reth_rpc_types::Transaction>> {
+        let block_by_hash = self.client().block_by_hash(hash)
+            .map_err(|_| EthApiError::UnknownBlockNumber)?
+            .ok_or(EthApiError::UnknownBlockNumber)?;
```

Comment:
is this correct?
I think this should return null if the transaction does not exist?